### PR TITLE
Plug some leaks in curl's proxy handling

### PR DESCRIPTION
**Pull request by carlosmn**
I was trying to reproduce the segfault we see in Travis when talking to the proxy, but instead I found a few leaks.

### CODE REVIEW

**Comment by ethomson** [comment]
Neat, thanks for fixing these.  Which segfault?  The one from with NTLM creds from #4317 and #4321?

**Comment by carlosmn** [comment]
I was looking at #4339 where the thread-unsafe suite seems to be failing somewhat reliably, but I've been unable to reproduce it.

**Comment by ethomson** [comment]
Oh, that's interesting, isn't it.  I hadn't seen that.

File: src/proxy.c [hunk 1]
```diff
@@ -31,3 +31,9 @@ int git_proxy_options_dup(git_proxy_options *tgt, const git_proxy_options *src)
 
 	return 0;
 }
+
+void git_proxy_options_free(git_proxy_options *opts)
```
Comment (pks-t) [inline src/proxy.c]: We usually call this kind of function `git_proxy_options_clear` instead of `free`. A `free` should also free the structure itself.

Comment (carlosmn) [reply src/proxy.c]: Sure, that makes sense. Done.

**Comment by ethomson** [comment]
Thanks @carlosmn !

**Pull request merged**